akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/caption/CaptionApp2x.scala

```diff
@@ -59,7 +65,23 @@ class CaptionApp2x(implicit val context: ActorContext) extends RightsManagementT
       }
     }
   }
+  def handle(msg: CaptionSubmitTranscriptPubMsg, liveMeeting: LiveMeeting, bus: MessageBus): Unit = {
+    val meetingId = liveMeeting.props.meetingProp.intId
+    def broadcastSuccessEvent(transcriptId: String, transcript: String, locale: String): Unit = {
+      val routing = Routing.addMsgToClientRouting(MessageTypes.BROADCAST_TO_MEETING, liveMeeting.props.meetingProp.intId, msg.header.userId)
+      val envelope = BbbCoreEnvelope(CaptionSubmitTranscriptSuccessEvtMsg.NAME, routing)
+      val header = BbbClientMsgHeader(CaptionSubmitTranscriptSuccessEvtMsg.NAME, liveMeeting.props.meetingProp.intId, msg.header.userId)
+
+      val body = CaptionSubmitTranscriptSuccessEvtMsgBody(transcriptId, transcript, locale)
+      val event = CaptionSubmitTranscriptSuccessEvtMsg(header, body)
+      val msgEvent = BbbCommonEnvCoreMsg(envelope, event)
+      bus.outGW.send(msgEvent)
+    }
+    CaptionDAO.insertOrUpdateAudioCaption(msg.body.transcriptId, meetingId, msg.header.userId,

```

Comment:
Consider renaming this function if it will be used by TypedCaptions too.
And then your will need to inform the `captionType` as a param.